packages/core/src/authenticatedFetch/fetchFactory.spec.ts

```diff
@@ -468,4 +470,65 @@ describe("buildAuthenticatedFetch", () => {
     await sleep(200);
     expect(refreshCall.mock.calls[1][1]).toEqual("some rotated refresh token");
   });
+
+  it("emits the appropriate events when refreshing the token fails", async () => {
+    const mockedFetch = jest.requireMock("cross-fetch") as jest.Mock;
+    const mockedFreshener = mockTokenRefresher(mockDefaultTokenSet());
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    mockedFreshener.refresh = jest
+      .fn()
+      .mockRejectedValueOnce(
+        new OidcProviderError(
+          "Some error message",
+          "error_identifier",
+          "Some error description"
+        )
+      ) as any;
+    const mockEmitter = new EventEmitter();
+    const spiedEmit = jest.spyOn(mockEmitter, "emit");
+
+    await buildAuthenticatedFetch(mockedFetch, "myToken", {
+      refreshOptions: {
+        refreshToken: "some refresh token",
+        sessionId: "mySession",
+        tokenRefresher: mockedFreshener,
+        expiresIn: 0,
+        eventEmitter: mockEmitter,
+      },
+    });
+    await sleep(200);
+    expect(spiedEmit).toHaveBeenCalledTimes(2);
+    expect(spiedEmit).toHaveBeenCalledWith(EVENTS.SESSION_EXPIRED);
+    expect(spiedEmit).toHaveBeenCalledWith(
+      EVENTS.ERROR,
+      "error_identifier",
+      "Some error description"
+    );
+  });
+
+  it("emits the appropriate events when an unexpected response is received", async () => {
+    const mockedFetch = jest.requireMock("cross-fetch") as jest.Mock;
+    const mockedFreshener = mockTokenRefresher(mockDefaultTokenSet());
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Damnit, I already miss half these warnings because they are displayed before the coverage report, and I don't necessarily scroll up there, but it ESLint and Prettier tag team I'm utterly defeated.